src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -34,6 +33,32 @@ object SnowflakeTelemetry {
 
   private[snowflake] var output: ObjectNode = _
 
+  private var telemetryMessageSender: TelemetryMessageSender = new SnowflakeTelemetryMessageSender
+
+  private[snowflake] def setTelemetryMessageSenderForTest(sender: TelemetryMessageSender)
+  : TelemetryMessageSender = {
+    val oldSender = telemetryMessageSender
+    telemetryMessageSender = sender
+    oldSender
+  }
+
```

Comment:
Use `cachedSparkApplicationId` + `getSparkApplicationId` instead of `lazy val` to avoid caching an invalid spark application id.
The potential use case is:
Users use `Util.runQuery()` to execute a query before creating SparkSession. `runQuery()` creates a connection so it may not get a valid spark application id.